compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/nballerina/NBir.java

```diff
@@ -0,0 +1,894 @@
+        insns.forEach(insn -> arr.append(insn.getRecord()));
+        Map<String, Object> fields = new HashMap<>();
+        fields.put("label", label);
+        fields.put("insns", arr);
+        if (ppb) {
+            fields.put("onPanic", onPanic);
+        }
+        return ValueCreator.createRecordValue(ModuleGen.MODBIR, "BasicBlock", fields);
+    }
+}
+
+abstract class InsnBase {
+    public String name; //InsnName
+    public abstract BMap<BString, Object> getRecord();
+
+}
+
+class CondBranchInsn extends InsnBase {
+    public Register operand;
+    int ifTrue;
+    int ifFalse;
+
+    public CondBranchInsn(Register operand, int ifTrue, int ifFalse) {
+        this.operand = operand;
+        this.ifTrue = ifTrue;
+        this.ifFalse = ifFalse;
+    }
+
+    @Override
+    public BMap<BString, Object> getRecord() {
+        LinkedHashMap<String, Object> fields = new LinkedHashMap<>();
+        fields.put("operand", operand.getRecord());
+        fields.put("ifTrue", ifTrue);
+        fields.put("ifFalse", ifFalse);
+        return ValueCreator.createReadonlyRecordValue(ModuleGen.MODBIR, NBTypeNames.COND_BRANCH_INSN, fields);
+    }
+}
+
+class BranchInsn extends InsnBase {
+    int dest;
+
+    public BranchInsn(int dest) {
+        this.dest = dest;
+    }
+
+    @Override
+    public BMap<BString, Object> getRecord() {
+        LinkedHashMap<String, Object> fields = new LinkedHashMap<>();
+        fields.put("dest", dest);
+        return ValueCreator.createReadonlyRecordValue(ModuleGen.MODBIR, NBTypeNames.BRANCH_INSN, fields);
+    }
+}
+
+class IntArithmeticBinaryInsn extends InsnBase {
+    public String op;
+    public Register result;
+    public Operand[] operands = new Operand[2];
+    public Position position;
+
+    @Override
+    public BMap<BString, Object> getRecord() {
+        BMap<BString, Object> tmpRegVal = ValueCreator.createReadonlyRecordValue(ModuleGen.MODBIR, NBTypeNames.REGISTER,
+                new HashMap<>());
+        UnionType typ = TypeCreator.createUnionType(tmpRegVal.getType(), PredefinedTypes.TYPE_INT);
+        ArrayType arrTyp = TypeCreator.createArrayType(typ, 2);
+        BArray arr = ValueCreator.createArrayValue(arrTyp);
+        arr.add(0, operands[0].getOperand());
+        arr.add(1, operands[1].getOperand());
+        Map<String, Object> fields = new HashMap<>();
+        fields.put("op", new BmpStringValue(op));
+        fields.put("result", result.getRecord());
+        fields.put("position", position.getRecord());
+        fields.put("operands", arr);
+        return ValueCreator.createReadonlyRecordValue(ModuleGen.MODBIR, NBTypeNames.INT_ARITHMETIC_BINARY_INSN, fields);
+    }
+}
+
+class StringConcatInsn extends InsnBase {
+    public Register result;
+    public Operand[] operands = new Operand[2];
+
+    @Override
+    public BMap<BString, Object> getRecord() {
+        BMap<BString, Object> tmpRegVal = ValueCreator.createReadonlyRecordValue(ModuleGen.MODBIR, NBTypeNames.REGISTER,
```

Comment:
this is a general comment, lot of the names can be much simpler, eg here I would think `registerTy` would be a name.